### PR TITLE
Update the UI of authentication flow pages as per feedback

### PR DESCRIPTION
**Pull request by sanjibroy360**
## Notion card
[https://www.notion.so/saeloun/Signin-Signup-UI-Review-f6ec879d28824cbfb0868a83524ac685](https://www.notion.so/saeloun/Signin-Signup-UI-Review-f6ec879d28824cbfb0868a83524ac685)

## Summary
- Ensured that the gap between elements should be consistent.
- Update the UI of all error screens.
- Show validation errors only when the user tries to submit the form, not on blur.
- In mobile and iPad view we were not showing the Miru logo anywhere on the Sign-up and Sign-in page. I've added that as well.
- Ensured that the label and input-box's text should be aligned correctly.

## Preview
[https://www.loom.com/share/a702c7d2417f40ffaeaa52494cb91f7e](https://www.loom.com/share/a702c7d2417f40ffaeaa52494cb91f7e)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactor (non-breaking and retains same functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### Checklist:

- [x] I have manually tested all workflows
- [x] I have performed a self-review of my own code
- [x] I have annotated changes in the PR that are relevant to the reviewer
- [ ] I have added automated tests for my code

### CODE REVIEW

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

File: app/javascript/src/components/Authentication/ForgotPassword/index.tsx [hunk 1]
```diff
@@ -44,34 +44,55 @@ const ForgotPassword = () => {
...
 
                 return (
                   <Form>
                     <div className="field relative">
                       <InputField
                         id="email"
                         label="Enter your registered email ID"
+                        labelClassName="p-0"
                         name="email"
+                        inputBoxClassName={`p-3.75 ${
```
[the remark this answers is not in the document]
Comment (sanjibroy360) [reply app/javascript/src/components/Authentication/ForgotPassword/index.tsx:85]: Okay @prasanthchaduvula , I'll move the `inputBoxClassName` logic to `<InputField/>`. 

We are implementing the onChange function to clear validation errors as the user types. 
So now validation error messages will only be displayed when the user tries to submit the form(`onSubmit`) with invalid data and we'll clear those errors as the user types(`onChange`).

[the remark this answers is not in the document]
Comment (prasanthchaduvula) [reply app/javascript/src/components/Authentication/ForgotPassword/index.tsx:85]: Can you try adding this logic of `onChange` to `<InputField/>` itself?

[the remark this answers is not in the document]
Comment (sanjibroy360) [reply app/javascript/src/components/Authentication/ForgotPassword/index.tsx:85]: Okay

**Comment by supriya3105** [comment]
@prasanthchaduvula Please review. Would like the fixes to be part of release 0.16.0.

**Comment by prasanthchaduvula** [comment]
@sanjibroy360 all the changes completed?

File: app/javascript/src/components/Authentication/ForgotPassword/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (sanjibroy360) [reply app/javascript/src/components/Authentication/ForgotPassword/index.tsx]: Done

**Comment by sanjibroy360** [comment]
@prasanthchaduvula Two tasks remaining: update signup system specs and fix regex issue.

File: app/javascript/src/components/Authentication/ForgotPassword/index.tsx [hunk 2]
```diff
@@ -44,34 +44,48 @@ const ForgotPassword = () => {
...
+                  setFieldValue,
+                  setFieldError,
+                } = props;
 
                 return (
                   <Form>
                     <div className="field relative">
                       <InputField
+                        resetErrorOnChange
+                        hasError={errors.email && touched.email}
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Authentication/ForgotPassword/index.tsx:82]: Please add 
`resetErrorOnChange`,
`hasError`,
`setFieldError={setFieldError}`
`setFieldValue={setFieldValue}`
as a default value to `<InputField/>`

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

File: app/javascript/src/components/Authentication/ForgotPassword/index.tsx (hunk 2 above)
Comment (sanjibroy360) [reply app/javascript/src/components/Authentication/ForgotPassword/index.tsx:82]: @prasanthchaduvula , I'll set a default value of `true` for `resetErrorOnChange`. But we have to pass the following three props: `hasError, setFieldError, and setFieldValue`. 

Here `setFieldError` and `setFieldValue` are optional props and only need to be passed if we want to clear validation errors upon input change(when `resetErrorOnChange` is true).

Comment (sanjibroy360) [reply app/javascript/src/components/Authentication/ForgotPassword/index.tsx:81]: Done

**Review by prasanthchaduvula — APPROVED** [review]

**Comment by apoorv1316** [comment]
@sanjibroy360 We are getting 2 toasts in case of an invalid email/password. And all errors are shown below their respective fields but if we add an invalid name then the error message for it is shown via toaster and the same for invalid email/password. We should handle it in a consistent manner. cc @supriya3105 . Also, get the UI reviewed by Akash once.
Loom: https://www.loom.com/share/c8ff9b331239479685341683089834b3

**Comment by sanjibroy360** [comment]
> @sanjibroy360 We are getting 2 toasts in case of an invalid email/password. And all errors are shown below their respective fields but if we add an invalid name then the error message for it is shown via toaster and the same for invalid email/password. We should handle it in a consistent manner. cc @supriya3105 . Also, get the UI reviewed by Akash once. Loom: https://www.loom.com/share/c8ff9b331239479685341683089834b3

@apoorv1316 I've resolved these 2 issues.

## Preview
[https://youtu.be/frGQmxIFS5w](https://youtu.be/frGQmxIFS5w)

**Comment by apoorv1316** [comment]
@sanjibroy360 We are getting 2 toasts on entering an email that doesn't exist in DB on forgot password page. Please fix it too.

**Comment by supriya3105** [comment]
@sanjibroy360 Please check why the tests are failing . 

@prasanthchaduvula Lets review again after Sanjib fixes the tests.

**Comment by sanjibroy360** [comment]
> @sanjibroy360 We are getting 2 toasts on entering an email that doesn't exist in DB on forgot password page. Please fix it too.

@apoorv1316 

> @sanjibroy360 We are getting 2 toasts on entering an email that doesn't exist in DB on forgot password page. Please fix it too.

Done 
cc: @apoorv1316

**Comment by sanjibroy360** [comment]
@prasanthchaduvula now all tests are passing here.

**Review by prasanthchaduvula — APPROVED** [review]

**Pull request merged**